### PR TITLE
Addition Circle

### PR DESCRIPTION
**Pull request by taiking**
# Summary
Added setting to be able to change underlined background, like this
gif

### CODE REVIEW

File: Example/PopupViewController.swift [hunk 1]
```diff
@@ -75,7 +75,7 @@ class PopupViewController: UIViewController {
         tabItemViewWidthLabel.text = "Tab Item Width: \(String(format: "%.0f", Float(options.tabView.itemView.width)))"
 
         switch options.tabView.addition {
-        case .underline:
+        case .underline, .circle:
```
Comment (KazuhiroTamiwa) [inline Example/PopupViewController.swift]: rectangleもできるといいかもですね。

File: Sources/TabView.swift [hunk 2]
```diff
@@ -347,6 +347,15 @@ extension TabView {
             underlineView = UIView(frame: CGRect(x: itemView.frame.origin.x + options.underlineView.margin, y: itemView.frame.height, width: itemView.frame.width - options.underlineView.margin * 2, height: options.underlineView.height))
             underlineView.backgroundColor = options.underlineView.backgroundColor
             containerView.addSubview(underlineView)
+        case .circle:
+            let itemView = itemViews[currentIndex]
+            let height = min(itemView.titleLabel.font.pointSize + options.underlineView.padding * 2, itemView.frame.height)
+            underlineView = UIView(frame: CGRect(x: itemView.frame.origin.x + options.underlineView.margin, y: 0, width: itemView.frame.width - options.underlineView.margin * 2, height: height))
+            underlineView.layer.position.y = itemView.layer.position.y
+            underlineView.layer.cornerRadius = underlineView.frame.height / 2
```
Comment (KazuhiroTamiwa) [inline Sources/TabView.swift]: cornerRadiusも指定できるようになるとうれしいですね

File: Sources/SwipeMenuView.swift [hunk 3]
```diff
@@ -42,6 +43,9 @@ public struct SwipeMenuViewOptions {
 
             /// UndelineView side margin. Defaults to `0.0`.
             public var margin: CGFloat = 0.0
+            
+            /// UnderlineView top and bottom padding. Defaults to `5.0`.
+            public var padding: CGFloat = 5.0
```
Comment (yysskk) [inline Sources/SwipeMenuView.swift]: Create new options for circle as these are options for undelineView layout.

File: Sources/TabView.swift [hunk 4]
```diff
@@ -489,7 +498,7 @@ extension TabView {
     private func moveTabItem(index: Int, animated: Bool) {
 
         switch options.addition {
-        case .underline:
+        case .underline, .circle:
             animateUnderlineView(index: index, animated: animated, completion: nil)
```
Comment (yysskk) [inline Sources/TabView.swift]: Change the naming of this function as its meaning changes. (ex. `func animateAdditionView()` )

File: Sources/TabView.swift [hunk 5]
```diff
@@ -347,6 +347,15 @@ extension TabView {
             underlineView = UIView(frame: CGRect(x: itemView.frame.origin.x + options.underlineView.margin, y: itemView.frame.height, width: itemView.frame.width - options.underlineView.margin * 2, height: options.underlineView.height))
             underlineView.backgroundColor = options.underlineView.backgroundColor
             containerView.addSubview(underlineView)
+        case .circle:
```
Comment (yysskk) [inline Sources/TabView.swift]: Rename  `func setupUnderlineView()` as well

File: Sources/TabView.swift [hunk 6]
```diff
@@ -347,6 +347,15 @@ extension TabView {
             underlineView = UIView(frame: CGRect(x: itemView.frame.origin.x + options.underlineView.margin, y: itemView.frame.height, width: itemView.frame.width - options.underlineView.margin * 2, height: options.underlineView.height))
```
Comment (yysskk) [inline Sources/TabView.swift]: Rename to `var additionView: UIView` as well.

File: Sources/TabView.swift (hunk 5 above)
Comment (yysskk) [reply Sources/TabView.swift]: and Rename `func updateUnderlineViewPosition()`, `func resetUnderlineViewPosition()`, `func moveUnderlineView()`, too.

File: Example/PopupViewController.swift (hunk 1 above)
Comment (yysskk) [reply Example/PopupViewController.swift]: @KazuhiroTamiwa For what purpose does it use?

File: Sources/TabView.swift [hunk 7]
```diff
@@ -284,7 +284,7 @@ open class TabView: UIScrollView {
                     containerView.topAnchor.constraint(equalTo: self.topAnchor),
                     leftMarginConstraint,
                     containerView.widthAnchor.constraint(equalToConstant: containerWidth),
-                    containerView.heightAnchor.constraint(equalToConstant: options.height - options.underlineView.height)
+                    containerView.heightAnchor.constraint(equalToConstant: options.height)
```
Comment (yysskk) [inline Sources/TabView.swift]:
```
switch style {
case .underline:
  containerView.heightAnchor.constraint(equalToConstant: options.height - options.additionView.underline.height)
case .circle, .none:
  containerView.heightAnchor.constraint(equalToConstant: options.height)
}
```

File: Sources/TabView.swift [hunk 8]
```diff
@@ -293,7 +293,7 @@ open class TabView: UIScrollView {
...
```
Comment (yysskk) [inline Sources/TabView.swift]: Same

File: Sources/TabView.swift [hunk 9]
```diff
@@ -305,7 +305,7 @@ open class TabView: UIScrollView {
                     containerView.topAnchor.constraint(equalTo: self.topAnchor),
                     leftMarginConstraint,
                     widthConstraint,
-                    containerView.heightAnchor.constraint(equalToConstant: options.height - options.underlineView.height)
+                    containerView.heightAnchor.constraint(equalToConstant: options.height)
```
Comment (yysskk) [inline Sources/TabView.swift]: Same

File: Sources/TabView.swift [hunk 10]
```diff
@@ -314,7 +314,7 @@ open class TabView: UIScrollView {
...
```
Comment (yysskk) [inline Sources/TabView.swift]: Same

**Pull request merged**